### DISCUSSION TITLE
Should we allow ERC20 tokens with 0 decimal value?

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#116] [General]
<sup>Original discussed between @smol-ninja and @andreivladbrg on Slack.</sup>

In `create` function, there is a check that revert if asset has 0 decimal:

https://github.com/sablier-labs/flow/blob/b5f70b6fae489d841aca38564099bae304ba82cf/src/SablierV2OpenEnded.sol#L487-L490

Decimal value plays a key role in normalising the amounts into 18 decimal format.

https://github.com/sablier-labs/flow/blob/b5f70b6fae489d841aca38564099bae304ba82cf/src/SablierV2OpenEnded.sol#L312-L322

However, since the decimal value is not used in the denominator, it does not seem to create any calculation issues. I think that we should allow 0 decimal tokens, regardless of their demand, because fundamentally they are no different than any other ERC20 token.

Would love to hear your opinion [@sablier-labs/engineers](https://github.com/orgs/sablier-labs/teams/engineers).

**Comment by smol-ninja** [comment]
@andreivladbrg do you see any problem with 0 decimal? Should we go ahead and remove 0 decimal check?

**Reply by andreivladbrg** [reply]
sorry for taking longer to answer to this

after thinking more about this i am also of the opinion that it is safe to use assets with 0 decimals

**Comment by andreivladbrg** [comment]
Issue created: https://github.com/sablier-labs/flow/issues/144
